ui/perfherder/alerts/StatusDropdown.jsx

```diff
@@ -73,6 +78,45 @@ export default class StatusDropdown extends React.Component {
     };
   };
 
+  getNumberOfNonWorkingDays(createdAt, dueDate) {
+    let count = 0;
+    const curDate = new Date(createdAt.getTime());
+    const sunday = 0;
+    const saturday = 6;
+
+    curDate.setHours(0, 0, 0, 0);
+    dueDate.setHours(0, 0, 0, 0);
+
+    if (curDate.getDay() === sunday) {
+      return 1;
+    }
+
+    if (curDate.getDay() === saturday) {
+      return 2;
+    }
+
+    if (dueDate.getDay() === saturday || dueDate.getDay === sunday) {
+      this.isWeekend = true;
+      return 2;
+    }
+
+    while (curDate <= dueDate) {
```

Comment:
I was thinking we can simplify this method. Can you explain why this `while` is mandatory? To me it seems that the first part of this method resolves the issue of finding the number of days needed to shift the `dueDate` and this part is no longer needed. Please correct me if I'm not understanding this correctly.

Reply:
If you fix the previous mistake (`dueDate.getDay === sunday`) and add `    if (curDate.getDay() === friday) {
      return 2;
    }` you should be able to remove the while implementation.

Reply:
I would also suggest to have a variable for `dueDate.getDay()` so that you don't have to use `.getDay()` everywhere.